internal/provider/apppackage.go

```diff
@@ -39,10 +39,10 @@ func resourceAppFile() *schema.Resource {
 	return &schema.Resource{
 		Description:   "Create a tarball for an app. Generated app.conf's version will be automatically incremented when app content changes.",
 		CustomizeDiff: resourceAppPackageCustomDiff,
-		CreateContext: resourceAppPackageCreate,
 		// create/read/update end up doing the same work, so they use the same function
-		ReadContext:   resourceAppPackageCreate,
-		UpdateContext: resourceAppPackageCreate,
+		CreateContext: resourceAppPackageRead,
+		ReadContext:   resourceAppPackageRead,
+		UpdateContext: resourceAppPackageRead,
```

Comment:
This change simply makes this resource look more like a data source would look, with `Read` being the only operation. It was potentially unneeded after all, but isn't an invalid change so I've left it in place.